api/src/main/java/org/openmrs/module/datafilter/impl/ImplDataFilterListener.java

```diff
@@ -87,18 +87,21 @@ public boolean onEnableFilter(DataFilterContext filterContext) {
 			
 		} else if (filterContext.getFilterName().startsWith(ImplConstants.PROGRAM_BASED_FILTER_NAME_PREFIX)) {
 			Collection<String> userProgramRoleNames = new HashSet();
+			//Add '#####' place holder for non authenticated user
+			userProgramRoleNames.add("#####");
```

Comment:
Why not this as a one-liner with the original (better) comment?
```java
// Avoid a 'select IN ()' which would be an invalid query, in theory we expect no role to match #####
Collection<String> userProgramRoleNames = new HashSet<>(Arrays.asList("#####"));
```

Reply:
>`Collection<String> userProgramRoleNames = new HashSet<>(Arrays.asList("#####"));`

`userProgramRoleNames` should be final in nature since it will be used with the Stream filtering following.

Reply:
I'm suggesting a one-liner:
```java
Collection<String> userProgramRoleNames = new HashSet<>(Arrays.asList("#####"));
```
instead of a two-liner:
```java
Collection<String> userProgramRoleNames = new HashSet();
userProgramRoleNames.add("#####");
```

Reply:
Ahhh, yes for sure. My bad, I my mind I was referring to `allProgramRoleNames `. Let me apply the suggestion.